### ISSUE TITLE
[a11y]: Carbon multi select pill is not keyboard accessible

### ISSUE THREAD

**Issue by lawrencepobrien** [carbon-design-system/carbon#21113]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

v11.96.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Clear All icon for the selected value must receive Keyboard focus.
All the interactive elements in the webpage should be operable with the keyboard When interactive controls are inoperable with keyboard, User is unable to perform an action on the web page or complete a task. This defect impacts keyboard only users like blind screen reader users and users with motor disabilities.

https://github.ibm.com/catalog-management/content-mgmt-ui/issues/1294

### WCAG 2.1 Violation

2.1.1

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-multiselect--default

### Steps to reproduce

Select at least one element on the multiselect component and close the dropdown. The dismissable tag within the multiselect component is not accessible with the keyboard. Either the dismissable tag should be accessible via keyboard or it should be communicated to the user if another keyboard shortcut is used (in this case the escape key)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
